proxy/InkAPIInternal.h

```diff
@@ -281,6 +281,25 @@ typedef enum {
 
 class SslAPIHooks : public FeatureAPIHooks<TSSslHookInternalID, TS_SSL_INTERNAL_LAST_HOOK>
 {
+public:
+  void
+  append(TSSslHookInternalID id, INKContInternal *cont) override
+  {
+    if (cont->mutex == nullptr) {
+      FeatureAPIHooks<TSSslHookInternalID, TS_SSL_INTERNAL_LAST_HOOK>::append(id, cont);
+    } else {
+      ink_release_assert(!"Cannot have continuation with mutex on SSL hook");
```

Comment:
More discussion needs to be done before this PR is approved.  I am against having an API that accepts a mutex and then later asserts because a mutex was given.

Reply:
We already have an API that lets you _not_ pass a mutex then assert when it's not present.